cunumeric/deferred.py

```diff
@@ -1657,6 +1663,54 @@ def _diag_helper(
 
         task.execute()
 
+    @auto_convert([1, 2])
+    def put(self, indices: Any, values: Any) -> None:
+        if indices.base.kind == Future or indices.base.transformed:
+            indices = indices._convert_future_to_regionfield(
+                indices.base.kind == Future
+            )
+        if values.base.kind == Future or values.base.transformed:
+            values = values._convert_future_to_regionfield(
+                values.base.kind == Future
+            )
+        if self.base.kind == Future or self.base.transformed:
```

Comment:
You are creating a copy of `self` here and writing to that. This has the effect that any values written are not reflected on the original array:

```
x = cn.zeros(1)
cn.put(x, cn.arange(4), cn.ones(4), mode="clip")
print(x) # prints [0]
```

Possibly you want to do something like what you do in `set_item` https://github.com/nv-legate/cunumeric/blob/branch-22.10/cunumeric/deferred.py#L909-L912.

Reply:
you are right. Thank you!

Reply:
Now the same example fails for me with
```
Traceback (most recent call last):
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 440, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "/Users/mpapadakis/legion/bindings/python/build/lib/legion_top.py", line 255, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "a.py", line 10, in <module>
    cn.put(x, cn.arange(4), cn.ones(4), mode="clip")
  File "/Users/mpapadakis/legate.core/legate/core/context.py", line 232, in wrapper
    result = func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 143, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/module.py", line 3479, in put
    a.put(indices=indices, values=values, mode=mode)
  File "/Users/mpapadakis/legate.core/legate/core/context.py", line 232, in wrapper
    result = func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/coverage.py", line 113, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 143, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/array.py", line 2510, in put
    self._thunk.put(indices._thunk, values._thunk)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 117, in wrapper
    return func(*args, **kwargs)
  File "/Users/mpapadakis/cunumeric/cunumeric/deferred.py", line 1687, in put
    N = self.ndim
AttributeError: 'NoneType' object has no attribute 'ndim'
```

Reply:
Should be fixed now. I should have checked your example earlier